File: android/src/main/java/com/margelo/nitro/visioncameramlkit/TextRecognizerOptions+toDomain.kt
package com.margelo.nitro.visioncameramlkit

import com.visioncameramlkit.domain.models.ImagePreprocessingOptions
import com.visioncameramlkit.domain.models.TextRecognitionOptions

internal fun TextRecognizerOptions.toDomainTextRecognitionOptions(): TextRecognitionOptions =
  TextRecognitionOptions(
    language = language.toDomainLanguage(),
    invertColors = invertColors ?: false,
    scaleFactor = scaleFactor?.toFloat() ?: 1.0f,
  )

internal fun TextRecognizerOptions.toImagePreprocessingOptions(): ImagePreprocessingOptions =
  ImagePreprocessingOptions(
    invertColors = invertColors ?: false,
    scaleFactor = scaleFactor?.toFloat() ?: 1.0f,
    roi = roi.toDomainRegionOfInterest(),
  )

File: ios/Bridge/Hybrid/TextRecognizerOptions+toDomain.swift
import Foundation

extension TextRecognizerOptions {
  func toDomainTextRecognitionOptions() -> TextRecognitionOptions {
    return TextRecognitionOptions(
      language: language?.toDomainLanguage() ?? .latin,
      invertColors: invertColors ?? false,
      scaleFactor: CGFloat(scaleFactor ?? 1.0)
    )
  }

  func toImagePreprocessingOptions() -> ImagePreprocessingOptions {
    return ImagePreprocessingOptions(
      invertColors: invertColors ?? false,
      scaleFactor: CGFloat(scaleFactor ?? 1.0),
      roi: roi?.toDomainRegionOfInterest()
    )
  }
}
